tests/configuration/test_configuration.py

```diff
@@ -368,3 +369,19 @@ def test_cannot_load_modular_configuration_with_broken_reference(tmp_path):
               - {secondary_file}
             """
         )
+
+
+def test_merge_configuration_expanduser():
```

Comment:
Maybe
```suggestion
def test_resolve_external_references_into_home_directory():
```

In any case: Good catch on using `Path.expanduser()` over `os.expanduser()` here.